runtime/parachains/src/configuration/migration.rs

```diff
@@ -16,10 +16,319 @@
+		pub pvf_voting_ttl: SessionIndex,
+		pub minimum_validation_upgrade_delay: BlockNumber,
+	}
+
+	impl<BlockNumber: Default + From<u32>> Default for HostConfiguration<BlockNumber> {
+		fn default() -> Self {
+			Self {
+				group_rotation_frequency: 1u32.into(),
+				chain_availability_period: 1u32.into(),
+				thread_availability_period: 1u32.into(),
+				no_show_slots: 1u32.into(),
+				validation_upgrade_cooldown: Default::default(),
+				validation_upgrade_delay: Default::default(),
+				code_retention_period: Default::default(),
+				max_code_size: Default::default(),
+				max_pov_size: Default::default(),
+				max_head_data_size: Default::default(),
+				parathread_cores: Default::default(),
+				parathread_retries: Default::default(),
+				scheduling_lookahead: Default::default(),
+				max_validators_per_core: Default::default(),
+				max_validators: None,
+				dispute_period: 6,
+				dispute_post_conclusion_acceptance_period: 100.into(),
+				dispute_max_spam_slots: 2,
+				dispute_conclusion_by_time_out_period: 200.into(),
+				n_delay_tranches: Default::default(),
+				zeroth_delay_tranche_width: Default::default(),
+				needed_approvals: Default::default(),
+				relay_vrf_modulo_samples: Default::default(),
+				max_upward_queue_count: Default::default(),
+				max_upward_queue_size: Default::default(),
+				max_downward_message_size: Default::default(),
+				ump_service_total_weight: OldWeight(Default::default()),
+				max_upward_message_size: Default::default(),
+				max_upward_message_num_per_candidate: Default::default(),
+				hrmp_sender_deposit: Default::default(),
+				hrmp_recipient_deposit: Default::default(),
+				hrmp_channel_max_capacity: Default::default(),
+				hrmp_channel_max_total_size: Default::default(),
+				hrmp_max_parachain_inbound_channels: Default::default(),
+				hrmp_max_parathread_inbound_channels: Default::default(),
+				hrmp_channel_max_message_size: Default::default(),
+				hrmp_max_parachain_outbound_channels: Default::default(),
+				hrmp_max_parathread_outbound_channels: Default::default(),
+				hrmp_max_message_num_per_candidate: Default::default(),
+				ump_max_individual_weight: OldWeight(
+					frame_support::weights::constants::WEIGHT_PER_MILLIS.ref_time() * 20,
+				),
+				pvf_checking_enabled: false,
+				pvf_voting_ttl: 2u32.into(),
+				minimum_validation_upgrade_delay: 2.into(),
+			}
+		}
+	}
+}
+
+pub fn migrate_to_v3<T: Config>() -> Weight {
+	// Unusual formatting is justified:
+	// - make it easier to verify that fields assign what they supposed to assign.
+	// - this code is transient and will be removed after all migrations are done.
+	// - this code is important enough to optimize for legibility sacrificing consistency.
+	#[rustfmt::skip]
+	let translate =
+		|pre: v2::HostConfiguration<BlockNumberFor<T>>| ->
+configuration::HostConfiguration<BlockNumberFor<T>>
+	{
+		super::HostConfiguration {
+max_code_size                            : pre.max_code_size,
```

Comment:
indentation is wrong?

Reply:
Yes, this is justified by the comment above the function block:

```rust
	// Unusual formatting is justified:
	// - make it easier to verify that fields assign what they supposed to assign.
	// - this code is transient and will be removed after all migrations are done.
	// - this code is important enough to optimize for legibility sacrificing consistency.
```